components/pages/privacy/access/requests/index.jsx

```diff
@@ -39,35 +43,36 @@ import {
   TOS_PREDICATE,
 } from "../../../../../__testUtils/mockApp";
 import { getRequestorWebId } from "../../../../../src/models/consent/request";
-import { getProfileResource } from "../../../../../src/solidClientHelpers/resource";
 import Spinner from "../../../../spinner";
 
 const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 
 export default function ConsentShow() {
   const { session } = useSession();
   const { fetch } = session;
-  const router = useRouter();
-  const { requestVc } = router.query;
+  const redirectUrl = window.location.href;
   const bem = useBem(useStyles());
   const [consentRequest, setConsentRequest] = useState(null);
   const agentWebId = getRequestorWebId(consentRequest);
   const [agentDetails, setAgentDetails] = useState({});
   const { agentName, agentUrl, agentPolicy, agentTOS } = agentDetails || null;
```

Comment:
An alternative to having all these states would be to lift the loading logic up into a function that returns back data in the shape you need it, then you'd just have:

```
const [isLoading, setLoading] = useState(true); // default to loading
const [error, setError] = useState(null);
const [requestDetails, setRequestDetails] = useState(null);

useEffect(() => {
  // You could also wrap this in a `Promise.race()` with a `sleep` to provide a timeout error.
  fetchDetails(window.location.href).then(
    (details) => setRequestDetails(details),
    (err) => setError(err)
  ).finally(() => setLoading(false));
});
```

And then have the `fetchDetails` method handle loading everything required, returning back an object of `{ consentRequest, agentWebId, agentDetails }`, as I don't think any of the individual state would change, as it's all predicated on the `window.location.href`

This would reduce the amount of back-and-forth react has to do managing the state, and make your code more testable.